### ISSUE TITLE
error during start (unknown paramsetDescription)

### ISSUE THREAD

**Issue by holgerimbery** [hobbyquaker/hm2mqtt.js#12]
Hello hobbyquaker,
not really sure, if it is an issue or an user error.
Any hint ?

```
hm2mqtt -m mqtt://192.168.1.90 -p xxxxxxxx -u admin -a 192.168.1.98 -j /home/pi/names.json
2017-04-20 15:45:31.759 <info>  hm2mqtt 1.0.2 starting
2017-04-20 15:45:31.771 <info>  mqtt trying to connect mqtt://192.168.1.90
2017-04-20 15:45:31.820 <info>  loading name table /home/pi/names.json
2017-04-20 15:45:31.874 <info>  rpc rfd > init [ 'xmlrpc_bin://192.168.1.90:2127', 'hm2mqtt_rfd' ]
2017-04-20 15:45:31.904 <info>  rpc hmip > init [ 'http://192.168.1.90:2126', 'hm2mqtt_hmip' ]
2017-04-20 15:45:32.419 <error> unknown paramsetDescription HM-RC-Key3-B/11/MAINTENANCE
2017-04-20 15:45:32.459 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.461 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.462 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.463 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.463 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.562 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.563 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.563 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.564 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.564 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.565 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.565 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.566 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.566 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.567 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:32.568 <error> unknown paramsetDescription HM-CC-RT-DN/29/CLIMATECONTROL_RT_TRANSCEIVER
2017-04-20 15:45:37.302 <info>  rfd got 304 devices and channels
/usr/lib/node_modules/hm2mqtt/index.js:614
    log.info(name, 'got', Object.keys(devices[name]).length, 'devices and channels');
                                 ^

TypeError: Cannot convert undefined or null to object
    at createParamsetQueue (/usr/lib/node_modules/hm2mqtt/index.js:614:34)
    at Timeout.getParamsetTimeout.(anonymous function).setTimeout [as _onTimeout] (/usr/lib/node_modules/hm2mqtt/index.js:805:13)
    at ontimeout (timers.js:380:14)
    at tryOnTimeout (timers.js:244:5)
    at Timer.listOnTimeout (timers.js:214:5)
```

**Comment by hobbyquaker** [comment]
seems to be a bug. will look into it asap

**Comment by hobbyquaker** [comment]
I catched this error, but I fear we have to investigate a little further. Could you pls try again with v1.0.3? And pls add the `-v debug` option.

**Comment by holgerimbery** [comment]
Thanks, that was fast. But ...

**Comment by holgerimbery** [comment]
sorry, wrong button :-)

it seems that you spotted the bug, but still no message in mosquitto


[logfile.txt](https://github.com/hobbyquaker/hm2mqtt.js/files/945143/logfile.txt)

**Comment by hobbyquaker** [comment]
Ok, I see two more issues here. I still have a Bug on fetching the Paramsets, no show-stopper, but have to fix that. The other thing is why you don't receive messages on mqtt - there is no successful connection to the Broker in your log. Have to catch more errors here to and adjust the log output, will do asap.
I must admit that I didn't test it with user/password auth, that could be a cause of your problem. Could you take a look into your mqtt brokers logfile when starting hm2mqtt?

**Comment by hobbyquaker** [comment]
I released v1.0.4 right now with improved mqtt logging. Could you try again with this version?

**Comment by holgerimbery** [comment]
first of all - thank you for your support !!

hm2mqtt is working, if I disable authentication, but not working with authentication enabled.

edit:
the problem is in index.js
line 56 - if you hardcode username and password here as:
    username : 'xxxxxx',
    password : 'yyyyyy'
it´s working


```
hm2mqtt -m mqtt://192.168.1.90 --mqtt-username admin --mqtt-password xxxxxxx --name hm --ccu-address 192.168.1.98 -j /home/pi/names.json -v debug
2017-04-21 05:58:18.912 <info>  hm2mqtt 1.0.4 starting
2017-04-21 05:58:18.922 <debug> loading paramsetDescriptions_192.168.1.98
2017-04-21 05:58:18.963 <info>  mqtt trying to connect mqtt://192.168.1.90
2017-04-21 05:58:19.009 <info>  loading name table /home/pi/names.json
2017-04-21 05:58:19.016 <debug> discover interfaces
2017-04-21 05:58:19.042 <debug> loading devices_192.168.1.98_rfd
2017-04-21 05:58:19.053 <debug> createIface rfd binrpc 2001
2017-04-21 05:58:19.067 <info>  rpc rfd > init [ 'xmlrpc_bin://192.168.1.90:2127', 'hm2mqtt_rfd' ]
2017-04-21 05:58:19.091 <debug> loading devices_192.168.1.98_hmip
2017-04-21 05:58:19.092 <debug> createIface hmip xmlrpc 2010
2017-04-21 05:58:19.097 <info>  rpc hmip > init [ 'http://192.168.1.90:2126', 'hm2mqtt_hmip' ]
2017-04-21 05:58:19.161 <warn>  mqtt offline
2017-04-21 05:58:19.162 <warn>  mqtt close
2017-04-21 05:58:19.190 <debug> rpc < system.listMethods [ 'hm2mqtt_rfd' ]
2017-04-21 05:58:19.233 <debug> rpc hmip < init ""
2017-04-21 05:58:19.250 <debug> rpc < listDevices [ 'hm2mqtt_rfd' ]
2017-04-21 05:58:19.254 <debug> > 304
2017-04-21 05:58:19.509 <debug> rpc < listDevices [ 'hm2mqtt_hmip' ]
2017-04-21 05:58:19.510 <debug> > 0
```
```
1492762198: New connection from 192.168.1.90 on port 1883.
1492762198: Socket error on client <unknown>, disconnecting.
1492762199: New connection from 192.168.1.90 on port 1883.
1492762199: Socket error on client <unknown>, disconnecting.
1492762200: New connection from 192.168.1.90 on port 1883.
1492762200: Socket error on client <unknown>, disconnecting.
1492762201: New connection from 192.168.1.90 on port 1883.
1492762201: Socket error on client <unknown>, disconnecting.
1492762202: New connection from 192.168.1.90 on port 1883.
1492762202: Socket error on client <unknown>, disconnecting.
1492762203: New connection from 192.168.1.90 on port 1883.
1492762203: Socket error on client <unknown>, disconnecting.
```

**Comment by hobbyquaker** [comment]
stupid me ;) mqttUser !== mqttUsername.... Should be fixed in v1.0.5

**Comment by holgerimbery** [comment]
it is fixed !! thank you !
